### PR TITLE
Steps cleanup

### PR DESCRIPTION
**Pull request by sstanculeanu**
modified stateMachine to implement Executor
now PollingHandler is used for stateMachine
removed error on Execute from Step interface
extra clean up

### CODE REVIEW

**Review by dragos-rebegea — DISMISSED** [review]

File: factory/ethElrondBridgeComponents.go [hunk 1]
```diff
@@ -550,47 +552,71 @@ func (components *ethElrondBridgeComponents) Start() error {
 func (components *ethElrondBridgeComponents) createAndStartEthToElrondStateMachine() error {
 	log := core.NewLoggerWithIdentifier(logger.GetOrCreate(ethToElrondName), ethToElrondName)
 
-	//TODO add a Start method on the state machine and move the creation of the state machine in the constructor
```
Comment (iulianpascalau) [inline factory/ethElrondBridgeComponents.go:553]: the TODO is still relevant: the creation of the state machine and the polling handler can now be moved inside the constructor, thus simplifying the Start method:
can adapt the TODO (for another PR)
or implement the suggestion :)

Valid for the next create and start function

File: testsCommon/stateMachine/stateMachineMock.go [hunk 2]
```diff
@@ -16,6 +16,10 @@ type StateMachineMock struct {
 	CurrentStep   core.Step
 }
 
+func (smm *StateMachineMock) IsInterfaceNil() bool {
```
Comment (iulianpascalau) [inline testsCommon/stateMachine/stateMachineMock.go:19]: missing `// IsInterfaceNil -`

File: factory/ethElrondBridgeComponents.go [hunk 3]
```diff
@@ -102,6 +102,8 @@ type ethElrondBridgeComponents struct {
 
 	mutClosableHandlers sync.RWMutex
 	closableHandlers    []io.Closer
+
+	pollingHandlers []PollingHandler
```
Comment (iulianpascalau) [inline factory/ethElrondBridgeComponents.go:106]: :+1: could have been aligned if L105 would have been deleted

File: factory/ethElrondBridgeComponents.go [hunk 4]
```diff
@@ -406,8 +419,9 @@ func (components *ethElrondBridgeComponents) createEthereumRoleProvider(args Arg
 	}
 
 	components.addClosableComponent(pollingHandler)
+	components.pollingHandlers = append(components.pollingHandlers, pollingHandler)
 
-	return pollingHandler.StartProcessingLoop()
+	return nil
```
Comment (iulianpascalau) [inline factory/ethElrondBridgeComponents.go:424]: :+1:

**Review by iulianpascalau — APPROVED** [review]

**Review by dragos-rebegea — APPROVED** [review]

**Pull request merged**